### ISSUE TITLE
Vector schema gives a failure

### ISSUE THREAD

**Issue by mangolas** [metosin/kekkonen#27]
If the schema is vector, kekkonen fails to process it

```clojure
(s/defschema Test1 {:a s/Int})
(s/defschema Test1Vec [Test1])

(p/defnk ^:command kekkonen-vec-test
  [ctx data :- domain/Test1Vec]
  (success {:ok 1}))
```

Calling through api with

```
[
  {
    "a": 0
  }
]
```

Gives
```
java.lang.AssertionError: Assert failed: (every? map? maps)
	at kekkonen.common$deep_merge.doInvoke(common.clj:18)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.core$update_in.doInvoke(core.clj:5923)
	at clojure.lang.RestFn.invoke(RestFn.java:467)
	at kekkonen.common$deep_merge_from_to.invoke(common.clj:46)
	at kekkonen.common$deep_merge_to_from.invoke(common.clj:49)
	at clojure.core.protocols$iter_reduce.invoke(protocols.clj:49)
	at clojure.core.protocols$fn__6510.invoke(protocols.clj:112)
	at clojure.core.protocols$fn__6452$G__6447__6465.invoke(protocols.clj:13)
	at clojure.core$reduce.invoke(core.clj:6519)
	at kekkonen.ring$eval29003$ring_handler__29012$fn__29015$fn__29022.invoke(ring.clj:130)
	at clojure.core$some_fn$sp2__6927.invoke(core.clj:7148)
	at ring.middleware.http_response$wrap_http_response$fn__30226.invoke(http_response.clj:8)
	at kekkonen.middleware$wrap_not_found$fn__30370.invoke(middleware.clj:113)
	at ring.middleware.format_params$wrap_format_params$fn__29976.invoke(format_params.clj:117)
	at ring.middleware.format_params$wrap_format_params$fn__29976.invoke(format_params.clj:119)
	at ring.middleware.format_params$wrap_format_params$fn__29976.invoke(format_params.clj:119)
	at ring.middleware.format_params$wrap_format_params$fn__29976.invoke(format_params.clj:119)
	at ring.middleware.format_params$wrap_format_params$fn__29976.invoke(format_params.clj:119)
	at kekkonen.middleware$wrap_exceptions$fn__30358.invoke(middleware.clj:69)
	at ring.middleware.format_response$wrap_format_response$fn__30130.invoke(format_response.clj:183)
	at kekkonen.middleware$wrap_keyword_keys$fn__30367.invoke(middleware.clj:105)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__30262.invoke(keyword_params.clj:35)
	at ring.middleware.nested_params$wrap_nested_params$fn__30312.invoke(nested_params.clj:84)
	at ring.middleware.params$wrap_params$fn__30338.invoke(params.clj:64)
	at backend.handler$create_handler$fn__47863.invoke(handler.clj:37)
	at buddy.auth.middleware$wrap_authentication$fn__32884.invoke(middleware.clj:38)
	at ring.middleware.session$wrap_session$fn__31355.invoke(session.clj:102)
	at backend.handler$create_handler$fn__47865.invoke(handler.clj:49)
	at metosin.ring.util.cache$wrap_cache$fn__24328.invoke(cache.clj:28)
	at aleph.http.server$handle_request$fn__22271$f__10710__auto____22272.invoke(server.clj:153)
	at clojure.lang.AFn.run(AFn.java:22)
	at io.aleph.dirigiste.Executor$Worker$1.run(Executor.java:62)
	at manifold.executor$thread_factory$reify__10592$f__10593.invoke(executor.clj:36)
	at clojure.lang.AFn.run(AFn.java:22)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by ikitommi** [comment]
Oh, just now I got this. It's bad. will, fix.

**Comment by mangolas** [comment]
Great, thanks!

**Comment by ikitommi** [comment]
should be fixed in `[metosin/kekkonen "0.3.0-20160518.060840-3"]`.
